fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/MessageHelpers.java

```diff
@@ -174,6 +174,34 @@ public static Descriptors.FieldDescriptor findFieldDescriptorOnMessageByOrdinal(
         return message.getDescriptorForType().getFields().get(fieldOrdinal);
     }
 
+    public static boolean compareMessageEquals(@Nonnull Object o1, @Nonnull Object o2) {
+        if (!(o1 instanceof Message) || !(o2 instanceof Message)) {
+            return false;
+        }
+        MessageOrBuilder m1 = (MessageOrBuilder) o1;
+        MessageOrBuilder m2 = (MessageOrBuilder) o2;
+
+        if (m1.getDescriptorForType().getFields().size() != m2.getDescriptorForType().getFields().size()) {
```

Comment:
If `m1.getDescriptorForType()` and `m2.getDescriptorForType()` are the same, can we just return `equals`?

Reply:
That is true.